org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/AcquireRequestHandler.java

```diff
@@ -66,36 +69,36 @@ public AcquireRequest process(AcquireRequest bean) throws EventException {
 
 			bean.setStatus(Status.COMPLETE);
 			bean.setMessage(null);
-		} catch (ScanningException | InterruptedException e) {
+		} catch (Exception e) {
+			// if an exception occurs set the bean status to failed and add the
+			// exception message as the bean's message.
 			e.printStackTrace();
+			logger.error("Cannot acquire data for detector " + getBean().getDetectorName(), e);
 			bean.setStatus(Status.FAILED);
 			bean.setMessage(e.getMessage());
-			logger.error("Cannot acquire data for detector " + getBean().getDetectorName(), e);
-			throw new EventException(e);
 		}
 		
 		return bean;
 	}
 	
-	private IRunnableDevice<?> createRunnableDevice(AcquireRequest request) throws ScanningException, EventException {
+	private IRunnableDevice<?> createRunnableDevice(AcquireRequest request) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)